clang/include/clang/Basic/DiagnosticCASKinds.td

```diff
@@ -42,8 +42,7 @@ def remark_compile_job_cache_hit : Remark<
   "compile job cache hit for '%0' => '%1'">, InGroup<CompileJobCacheHit>;
 def remark_compile_job_cache_miss : Remark<
   "compile job cache miss for '%0'">, InGroup<CompileJobCacheMiss>;
-def remark_compile_job_cache_miss_result_not_found : Remark<
-  "compile job cache miss for '%0' (result not found: '%1')">,
-  InGroup<CompileJobCacheMiss>;
+def err_compile_job_cache_result_not_found : Error<
```

Comment:
I'd recommend rename to `err_compile_job_cache_failed`, "result not found" could be confused with "the key is missing".